addons/mail/models/ir_binary.py

```diff
@@ -0,0 +1,20 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import models
+
+
+class IrBinary(models.AbstractModel):
+    _inherit = ["ir.binary"]
+
+    def _find_record_check_access(self, record, access_token, field):
+        if record._name in ["res.partner", "mail.guest"] and field == "avatar_128":
+            current_partner, current_guest = self.env["res.partner"]._get_current_persona()
+            if current_partner or current_guest:
```

Comment:
Do we actually need to check those 2?
If message is accessible by public user without guest, it would be fine to show the avatar too I believe.